homeassistant/components/imap_email_content/sensor.py

```diff
@@ -53,7 +57,61 @@
 )
 
 
-def setup_platform(
+def register_issue(hass: HomeAssistant, config: ConfigType) -> None:
+    """Register an issue and suggest new config."""
+
+    name: str = config.get(CONF_NAME) or config[CONF_USERNAME]
+
+    unique_id = (
+        f"{name}_{config[CONF_USERNAME]}_{config[CONF_SERVER]}_{config[CONF_FOLDER]}"
+    )
+
+    if CONF_VALUE_TEMPLATE in config:
+        template: str = config[CONF_VALUE_TEMPLATE].template
+        template = template.replace("subject", 'trigger.event.data["subject"]')
+        template = template.replace("from", 'trigger.event.data["sender"]')
+        template = template.replace("date", 'trigger.event.data["date"]')
+        template = template.replace("body", 'trigger.event.data["text"]')
+    else:
+        template = '{{ trigger.event.data["subject"] }}'
+
+    template_sensor_config: ConfigType = {
+        "template": [
+            {
+                "trigger": [
+                    {
+                        "id": "custom_event",
+                        "platform": "event",
+                        "event_type": "imap_content",
+                        "event_data": {"sender": config[CONF_SENDERS][0]},
+                    }
+                ],
+                "sensor": [
+                    {
+                        "state": template,
+                        "name": name,
+                    }
+                ],
+            }
+        ]
+    }
+
+    ir.async_create_issue(
+        hass,
+        DOMAIN,
+        unique_id,
+        breaks_in_ha_version="2023.10.0",
+        is_fixable=False,
```

Comment:
Do you think we can make a repair flow that sets up the imap integration via an import step and then shows the template sensor config to paste into configuration.yaml?

Reply:
May be, it depends on my time available. One limitation is that the event filter can filter only one sender. The easy way would be to pick the first sender.

Reply:
The user would still need to add the config to configuration.yaml so we can allow needing some manual adjustments if we point that out.

Reply:
When we offer a repair flow, it does not show the issue dialog. So we could create an imap entry and fix that part, but cannot show the new YAML. The current dialog offers a link to set up an entry. We could also raise 2 issues: One dialog for the removal of the deprecated yaml. One dialog to offer a fix to setup an entry (but that will be a half fix). Together I think it is clearer to just keep the dialog. The yaml part should not be an issue as the current (depratated) config was already set up a configuration.yaml.

Reply:
Added a migration step using a repair flow